### PR TITLE
Add Trigger Warnings 

### LINKED ISSUE #64: Trigger warnings
Title

### PR DESCRIPTION
**Pull request by neelr**
Fixes #64

### CODE REVIEW

File: lib/main.ts [hunk 1]
```diff
@@ -350,7 +357,8 @@ export async function viewConfession(
   repository: Repository<Confession>,
   staging_ts: string,
   approved: boolean,
-  reviewer_uid: string
+  reviewer_uid: string,
+  tw_text = ""
```
Comment (anirudhb) [inline lib/main.ts:361]:
```suggestion
  tw_text: string | null = null,
```

File: lib/main.ts [hunk 2]
```diff
@@ -380,7 +388,9 @@ export async function viewConfession(
     console.log(`Publishing message...`);
     const published_message = await web.chat.postMessage({
       channel: confessions_channel,
-      text: sanitize(`*${record.id}*: ${record.text}`),
+      text: sanitize(
+        `*${record.id}*: ${tw_text == "" ? record.text : tw_text}`
```
Comment (anirudhb) [inline lib/main.ts:392]:
```suggestion
        `*${record.id}*: ${tw_text ?? record.text}`
```

File: pages/api/interaction_work.ts [hunk 3]
```diff
@@ -425,6 +458,39 @@ You are not the original poster of the confession, so cannot reply anonymously.*
...
+        if (record === undefined) {
+          throw `Failed to find single Postgres record with staging_ts=${staging_ts}`;
+        }
+
+        // Reply in thread
+        const r = await web.chat.postMessage({
+          channel: confessions_channel,
+          text: sanitize(record.text),
+          thread_ts: record.published_ts,
+        });
```
Comment (anirudhb) [inline pages/api/interaction_work.ts:492]: The record should be fetched first so that the confession view is atomic

File: lib/main.ts (hunk 2 above)
Comment (anirudhb) [reply lib/main.ts:392]: Add something that indicates it's a TW, like `TW: blah blah, open thread to view confession`

**Review by anirudhb — CHANGES_REQUESTED** [review]
nits, also I think this fork is out of date(?)

File: pages/api/interaction_work.ts (hunk 3 above)
Comment (anirudhb) [reply pages/api/interaction_work.ts:492]: Also, the TW should be added to the DB (update the schema)

**Comment by neelr** [comment]
> nits, also I think this fork is out of date(?)

I thinkkk its up to date with the master branch, I pulled before working, and the latest commit is there

File: lib/main.ts [hunk 4]
```diff
@@ -283,6 +285,11 @@ export async function postStagingMessage(
       ...createStagingBlocks(id, sanitize(text)),
       new ActionsSection([
         new ButtonAction(new PlainText(":true: Approve"), "approve", "approve"),
+        new ButtonAction(
+          new PlainText(":angerydog: Approve with TW"),
+          "approve:tw",
+          "approve:tw"
+        ),
```
Comment (anirudhb) [inline lib/main.ts:292]: Mind moving this to the end so it doesn't break muscle memory

**Review by anirudhb — APPROVED** [review]
lgtm, need to test this sometime. Do you mind setting up your own prox2 instance to test this and DM me the results on slack?

**Comment by neelr** [comment]
I checked it in #botspam, should work and have all the bugs ironed out

**Comment by anirudhb** [comment]
Thanks so much for sticking with me on this! Ready to merge :)

**Pull request merged**